### ISSUE TITLE
X500V2 kit Jumps when switching to AltHold

### ISSUE THREAD

**Issue by farhangnaderi** [PX4/PX4-Autopilot#19046]
After trying to fly with the Holybro X500 V2 kit which flies well with Stabilized mode, I noticed that the drone jumps up and down when switching to Altitude mode. The kit is equipped with Pixhawk 5X. A log is also as below:

https://review.px4.io/plot_app?log=21beb351-f2bb-4767-b71f-725946098ec0

I found something which might be relevant [here](https://github.com/PX4/PX4-Autopilot/issues/11717) but not sure if it is the case!!!

**Comment by dagar** [comment]
Can you share a picture of this vehicle?

**Comment by farhangnaderi** [comment]
Sure, here you go: 

![57E35ABE-E62D-4780-A2E9-AA218ECEA873](https://user-images.githubusercontent.com/46557204/150217145-c73f98db-f35e-4c88-895d-e4053f64574d.jpeg)

If there is a possible question of how I have mounted the Pixhawk: 

![9B80D8EC-002A-4FDC-BAE7-D0CC794D9EFD](https://user-images.githubusercontent.com/46557204/150217332-c8ce072e-0a76-45cf-8e12-8e916a3e41cd.jpeg)

**Comment by dagar** [comment]
Can you try reducing the hover thrust quite a bit? `MPC_THR_HOVER 0.23`

Did you calibrate the ESCs?

**Comment by bresch** [comment]
Not sure if it's related, but both times before the spike in alt mode, the `local_position_setpoint.vz` is missing a couple of samples:
![DeepinScreenshot_select-area_20220120101440](https://user-images.githubusercontent.com/14822839/150309191-fe2506af-5b12-4d1f-8941-ed66dccf9e99.png)

**Comment by dagar** [comment]
It's the land detected ground contact.
![Screenshot from 2022-01-20 11-45-28](https://user-images.githubusercontent.com/84712/150383484-1008a75b-2b45-4146-b81d-13b6ed87f164.png)

**Comment by dagar** [comment]
The misconfigured `MPC_THR_HOVER` could probably also be blamed for the land detector issues (leading to throttle spikes).

@MaEtUgR could we ramp up to this more gradually? https://github.com/PX4/PX4-Autopilot/blob/a2260e53da48495644e128562ad80c1c276e7bef/src/modules/mc_pos_control/MulticopterPositionControl.cpp#L429

**Comment by MaEtUgR** [comment]
> The misconfigured MPC_THR_HOVER could probably also be blamed for the land detector issues (leading to throttle spikes).

Right, you saw that from the has_low_thrust flag being constantly true and hence it getting into ground contact all the time:
![image](https://user-images.githubusercontent.com/4668506/153248577-60a55188-a18a-4f83-9e2a-99b06e514d19.png)

**Comment by MaEtUgR** [comment]
From the data it looks like either the drone is completely overpowered/"overgeared" but since the motors, ESCs and props reportedly stayed the same that's unlikely. What seems more likely is a problem with the output scaling. So either ESC calibration gone wrong. The ESC calibration was reportedly done so that might be the issue or the PWM range configuration. If the ESCs support DSHOT and that gets enabled it would avoid such issues. Otherwise a fresh 1000 - 2000us PWM calibration on the ESCs if necessary with the motor test would solve that part and PWM parameter defaults the range configuration.

**Comment by farhangnaderi** [comment]
@MaEtUgR @bresch Thanks for the review, it might be my radio since I have a three-select for flight modes and Altitude is before Position mode exactly. 
@dagar Hover thrust less than 40% was really lazy to takeoff as I understood from today's flight test. 

I have calibrated ESCs again and flew today, these are the logs: 
https://review.px4.io/plot_app?log=50fac13b-cf22-4259-9dca-ceb6f916e17f
https://review.px4.io/plot_app?log=f2b50c49-a5e5-4aea-ae2d-17c74aa0a03a
https://logs.px4.io/plot_app?log=4bd7236b-c02f-4756-8dab-fa1acd4455f0
https://logs.px4.io/plot_app?log=f7b08d90-f681-49e5-a52f-feaecaf0ef27
https://review.px4.io/plot_app?log=6827a6d8-43cb-42b1-8f2a-107d5065d7a5
